llvm/tools/llvm-ir2vec/llvm-ir2vec.cpp

```diff
@@ -283,10 +284,24 @@ int main(int argc, char **argv) {
   if (Mode == TripletMode && Level.getNumOccurrences() > 0)
     errs() << "Warning: --level option is ignored in triplet mode\n";
 
-  // Parse the input LLVM IR file
+  // Parse the input LLVM IR file or stdin
   SMDiagnostic Err;
   LLVMContext Context;
-  std::unique_ptr<Module> M = parseIRFile(InputFilename, Err, Context);
+  std::unique_ptr<Module> M;
+
+  if (InputFilename == "-") {
```

Comment:
I'm pretty sure you don't need all this complicated logic.

`parseIRFile` should have logic for getting the underlying `MemoryBuffer` from STDIN if `-` is passed as the file name.

Reply:
Thanks! It works indeed.